daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Converter.scala

```diff
@@ -370,6 +378,15 @@ object Converter {
       case _ => Left(s"Expected SParty but got $v")
     }
 
+  def toParties(v: SValue): Either[String, OneAnd[Set, Ref.Party]] =
+    v match {
+      case SList(FrontStackCons(x, xs)) =>
+        NonEmptyList(x, xs.toList: _*).traverse(toParty(_)).map(toNonEmptySet(_))
```

Comment:
`OneAnd` is way more flexible for this kind of thing.

```suggestion
        OneAnd(x, xs).traverse(toParty(_)).map(oafp => /*oafp: OneAnd[FrontStack, Party]; some conversion here*/)
```

Reply:
ah good point, somehow it didn’t occur to me to use `OneAnd` for something other than `Set` here.